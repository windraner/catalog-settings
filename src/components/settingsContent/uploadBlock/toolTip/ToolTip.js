import React, { Component } from 'react';

import './toolTip.css';

export default class ToolTip extends Component {
  render() {
    const { text, isShow } = this.props;

    if(!isShow) {
      return null;
    }

    return (
      <div className="catalog-settings-content__upload-tool-tip">
        {text}
      </div>
    );
  }
}
