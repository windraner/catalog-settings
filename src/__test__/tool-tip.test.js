import React from 'react';
import ToolTip from './ToolTip';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<ToolTip />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<ToolTip text="test" isShow={false} />);
    expect(wrapper.type()).toBe(null);
    wrapper.setProps({isShow: true});
    expect(wrapper.find('.catalog-settings-content__upload-tool-tip').text()).toBe('test');
  });
  it('matches the snapshot', () => {
    const tree = shallow(<ToolTip text="test" isShow={false} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
