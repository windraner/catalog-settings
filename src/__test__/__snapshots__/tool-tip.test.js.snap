// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToolTip /> matches the snapshot 1`] = `""`;
